deviceshifu/pkg/deviceshifu/deviceshifu.go

```diff
@@ -180,11 +199,44 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
 			}
 		}
 
-		log.Printf("handling instruction '%v' to '%v'", handlerInstruction, *handlerEdgeDeviceSpec.Address)
-		resp, err := handlerHTTPClient.Get("http://" + *handlerEdgeDeviceSpec.Address + "/" + handlerInstruction)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusServiceUnavailable)
-			log.Printf("HTTP error" + err.Error())
+		var resp *http.Response
+		var httpErr error
+		reqType := r.Method
+
+		log.Printf("handling instruction '%v' to '%v' with request type %v", handlerInstruction, *handlerEdgeDeviceSpec.Address, reqType)
+
+		if reqType == http.MethodGet {
```

Comment:
Use switch instead of if else?

Reply:
I do not recommend switch case here as in the future we may need to make the conditional statement more complex.